### PR TITLE
feat: Create table calculations_v1 with identity column

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+CREATE TABLE IF NOT EXISTS {CATALOG_NAME}.{WHOLESALE_INTERNAL_DATABASE_NAME}.calculations_v1
+(
+    calculation_id STRING NOT NULL,
+    calculation_type STRING NOT NULL,
+    calculation_period_start TIMESTAMP NOT NULL,
+    calculation_period_end TIMESTAMP NOT NULL,
+    {DATABRICKS-ONLY}calculation_version BIGINT GENERATED ALWAYS AS IDENTITY,
```
Comment (energinet-ajw) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql:7]: Why is it autogenerated and why only databricks?

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql:7]: It turns out that we still get duplicate versions, so with this new table we can ensure unique values:
https://www.databricks.com/blog/2022/08/08/identity-columns-to-generate-surrogate-keys-are-now-available-in-a-lakehouse-near-you.html

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql:7]: Unfortunately, it doesn't work without Databricks. Halfspace mentioned today that a modification to spark conf might fix it. I will talk to them again as soon as possible.

Comment (energinet-ajw) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql:7]: Ok. Good to know. In the review I meant that a comment in the SQL would be nice.

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+CREATE TABLE IF NOT EXISTS {CATALOG_NAME}.{WHOLESALE_INTERNAL_DATABASE_NAME}.calculations_v1
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202409031600_create_table__internal__calculations_v1.sql:1]: What is the plan? Are you going to migrate existing calculations to this table?
If so, then what happens to the existing versions?

**Review by johevemi — APPROVED** [review]

**Pull request merged**